test/integration/discovery.go

```diff
@@ -306,32 +309,165 @@ var _ = Describe("Discover Cluster", func() {
+			if pod == nil {
+				framework.Failf("Failed to find a pod for deployment: %s", dep.Name)
+			}
+
+			nodeName = pod.Spec.NodeName
+			zoneNode, err = updateNodeWithLabel(kubeClient, zoneLabelKey, regionLabelKey, zoneLabelValue, regionLabelValue, nodeName)
+
+			entityDTOs, _, err := discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
+			framework.ExpectNoError(err, "Failed completing discovery of test cluster")
+
+			podFullName = namespace + "/" + pod.Name
+
+			podEntityDTO, nodeEntityDTO = getPodNodeDTO(entityDTOs, podFullName)
+
+			//Verify zone/region label in the commodity list
+			if validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_LABEL, commodityZoneValue) {
+				framework.Failf("Zone label in node for pod with PV is not honored")
+			}
+			if validateBuyerSellerCommodity(podEntityDTO, nodeEntityDTO, nodeName, proto.CommodityDTO_LABEL, commodityRegionValue) {
+				framework.Failf("Region label in node for pod with PV is not honored")
+			}
+
+		})
+		AfterEach(func() {
+			//Teardown remove both labels
+			fmt.Printf("INSIDE PAPARAZZI")
+			deleteNode := deleteTopologyLabels(zoneNode, zoneLabelKey, kubeClient)
+			_ = deleteTopologyLabels(deleteNode, regionLabelKey, kubeClient)
+		})
+	})
+
 	// TODO: this particular Describe is currently used as the teardown for this
 	// whole test (not the suite).
 	// This will work only if run sequentially. Find a better way to do this.
 	Describe("test teardowon", func() {
 		It(fmt.Sprintf("Deleting framework namespace: %s", namespace), func() {
+			fmt.Printf("INSIDE MAIN PAPARAZZI")
 			f.AfterEach()
 		})
 	})
 })
 
-func validateBuyerSellerCommodity(entityDTOs []*proto.EntityDTO, nodeName string, podName string,
-	commodityType proto.CommodityDTO_CommodityType, commodityValue string) bool {
+func updateNodeWithLabel(kubeClient *kubeclientset.Clientset, zoneLabelKey string, regionLabelKey string,
+	zoneLabelValue string, regionLabelValue string, nodeName string) (*corev1.Node, error) {
+	var err error
+	var zoneNode *corev1.Node
+	zoneNode, err = kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to retrieve node with error %s", err)
+	}
+	zoneNode.Labels[zoneLabelKey] = zoneLabelValue //Label node with zone labels
+	zoneNode.Labels[regionLabelKey] = regionLabelValue
+	zoneNode, err = kubeClient.CoreV1().Nodes().Update(context.TODO(), zoneNode, metav1.UpdateOptions{})
+	if err != nil {
+		framework.Failf("Failed to update node %s with zone label: %s %s", zoneNode, zoneLabelKey+":"+zoneLabelValue, err)
+	}
+	return zoneNode, err
+}
 
-	var buyerRegsitered bool
-	var sellerRegistered bool
+func deleteTopologyLabels(delNode *corev1.Node, zoneKey string, kubeClient *kubeclientset.Clientset) *corev1.Node {
+	//Delete topology label from node
+	for label := range delNode.Labels {
+		if strings.Contains(label, zoneKey) {
+			delete(delNode.Labels, label)
```

Comment:
@irfanurrehman in updated code as I am using this function in common teardown I am iterating over labels to avoid error where label is not present in the node